File: submissions/cpp/289.cpp
class Solution {
public:
    void gameOfLife(vector<vector<int>> &board) {
        vector<vector<int>> ret(board);
        int m = board.size(), n = m == 0 ? 0 : board[0].size();
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                int &&neighbour = 0;
                for (int I = max(0, i - 1); I < min(m, i + 2); ++I) {
                    for (int J = max(0, j - 1); J < min(n, j + 2); ++J) {
                        if (I == i && J == j)
                            continue;
                        neighbour += board[I][J] & 1;
                    }
                }
                if (board[i][j] == 0) {
                    if (neighbour == 3)
                        ret[i][j] = 1;
                } else {
                    if (neighbour < 2 || neighbour > 3)
                        ret[i][j] = 0;
                }
            }
        }
        board = ret;
    }
};

File: LeetCode/python3/289.py
class Solution:
    def gameOfLife(self, board):
        """
        :type board: List[List[int]]
        :rtype: void Do not return anything, modify board in-place instead.
        """
        ret = copy.deepcopy(board)
        m = len(board)
        n = len(board[0]) if m > 0 else 0
        for i in range(m):
            for j in range(n):
                neighbour = 0
                for I in range(max(0, i - 1), min(m, i + 2)):
                    for J in range(max(0, j - 1), min(n, j + 2)):
                        neighbour += board[I][J] if I != i or J != j else 0
                if neighbour == 3 and board[i][j] == 0:
                    ret[i][j] = 1
                elif (neighbour < 2 or neighbour > 3) and board[i][j] == 1:
                    ret[i][j] = 0
        for i in range(m):
            for j in range(n):
                board[i][j] = ret[i][j]